frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -1,81 +1,192 @@
-import baseAPIClient from "./BaseAPIClient";
-import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
 import {
-  getLocalStorageObjProperty,
-  setLocalStorageObjProperty,
-} from "../utils/LocalStorageUtils";
-
+  FetchResult,
+  MutationFunctionOptions,
+  OperationVariables,
+} from "@apollo/client";
+import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
 import { AuthenticatedUser } from "../types/AuthTypes";
+// rest {
+// import baseAPIClient from "./BaseAPIClient";
+// import {
+//   getLocalStorageObjProperty,
+//   setLocalStorageObjProperty,
+// } from "../utils/LocalStorageUtils";
+// } rest
+// graphql {
+import { setLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+// } graphql
+
+// graphql {
+type LoginFunction = (
+  options?:
+    | MutationFunctionOptions<{ login: AuthenticatedUser }, OperationVariables>
+    | undefined,
+) => Promise<
+  FetchResult<
+    { login: AuthenticatedUser },
+    Record<string, unknown>,
+    Record<string, unknown>
+  >
+>;
 
 const login = async (
   email: string,
   password: string,
-): Promise<AuthenticatedUser> => {
+  loginFunction: LoginFunction,
+): Promise<AuthenticatedUser | null> => {
+  let user: AuthenticatedUser = null;
   try {
-    const { data } = await baseAPIClient.post(
-      "/auth/login",
-      { email, password },
-      { withCredentials: true },
-    );
-    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(data));
-    return data;
-  } catch (error) {
-    return null;
+    const result = await loginFunction({ variables: { email, password } });
+    user = result.data?.login ?? null;
+    if (user) {
+      localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(user));
+    }
+  } catch (e: unknown) {
+    // eslint-disable-next-line no-alert
+    window.alert("Failed to login");
```

Comment:
A small nitpick: I would leave the try-catch and the alert in the component since it's related to the UI. Don't need to address this right now though, I'll make a note to change it the next time we touch this section of the code 🙂